src/renderer/kube-object.store.ts

```diff
@@ -132,10 +155,30 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
     }
   }
 
-  async loadSelectedNamespaces(): Promise<void> {
-    const { namespaceStore } = await import("./components/+namespaces/namespace.store");
+  @action
+  mergeItems(partialItems: T[], { replace = false, updateStore = true, sort = true, filter = true } = {}): T[] {
+    let items = partialItems;
+
+    if (!replace) {
+      items = this.items.toJS();
+
+      partialItems.forEach(item => {
+        const index = items.findIndex(i => i.getId() === item.getId());
+
+        if (index < 0) items.push(item); // add
+        else items[index] = item; // update
+      });
+    }
+
+    if (filter) items = this.filterItemsOnLoad(items);
+    if (sort) items = this.sortItems(items);
+    if (updateStore) this.items.replace(items);
+
+    return items;
+  }
 
-    return this.loadAll(namespaceStore.getContextNamespaces());
+  async loadContextNamespaces(): Promise<void> {
```

Comment:
My confusion with the naming is that this sounds like it loads namespaces when it's actually loading items found in the namespaces. `loadAllFromContextNamespaces`? Or `loadAllFromContextNs`?

Reply:
Updated to `loadAllFromContextNamespaces()`